File: app/src/main/java/com/arcao/geocaching4locus/base/util/PreferenceUtil.java
package com.arcao.geocaching4locus.base.util;

import android.content.SharedPreferences;

import timber.log.Timber;

public class PreferenceUtil {
    public static int getParsedInt(SharedPreferences preferences, String key, int defaultValue) {
        String value = preferences.getString(key, null);
        if (value == null)
            return defaultValue;

        try {
            return Integer.parseInt(value);
        } catch (NumberFormatException e) {
            Timber.e(e);
            return defaultValue;
        }
    }

    public static double getParsedDouble(SharedPreferences preferences, String key, double defaultValue) {
        String value = preferences.getString(key, null);
        if (value == null)
            return defaultValue;

        try {
            return Double.parseDouble(value);
        } catch (NumberFormatException e) {
            Timber.e(e);
            return defaultValue;
        }
    }

    public static float getParsedFloat(SharedPreferences preferences, String key, float defaultValue) {
        String value = preferences.getString(key, null);
        if (value == null)
            return defaultValue;

        try {
            return Float.parseFloat(value);
        } catch (NumberFormatException e) {
            Timber.e(e);
            return defaultValue;
        }
    }

}

File: app/src/main/java/com/arcao/geocaching4locus/base/util/PreferenceUtil.kt
package com.arcao.geocaching4locus.base.util

import android.content.SharedPreferences

@Suppress("NOTHING_TO_INLINE")
inline fun SharedPreferences.getParsedInt(key: String, defValue: Int) =
    getString(key, null)?.toIntOrNull() ?: defValue

@Suppress("NOTHING_TO_INLINE")
inline fun SharedPreferences.getParsedDouble(key: String, defValue: Double) =
    getString(key, null)?.toDoubleOrNull() ?: defValue

@Suppress("NOTHING_TO_INLINE")
inline fun SharedPreferences.getParsedFloat(key: String, defValue: Float) =
    getString(key, null)?.toFloatOrNull() ?: defValue
